executables/collector/src/main/kotlin/de/cyface/collector/handler/MeasurementHandler.kt

```diff
@@ -0,0 +1,573 @@
+     */
+    private val mongoClient: MongoClient
+
+    /**
+     * The maximum number of `Byte`s which may be uploaded.
+     */
+    private val payloadLimit: Long
+
+    /**
+     * Creates a fully initialized instance of this class.
+     *
+     * @param config the configuration setting required to start the HTTP server
+     */
+    constructor(config: Config) : this(config.database, config.authProvider, config.measurementLimit) {}
+
+    /**
+     * Creates a fully initialized instance of this class.
+     *
+     * @param mongoClient The client to use to access the Mongo database.
+     * @param authProvider An auth provider used by this server to authenticate against the Mongo user database
+     * @param payloadLimit The maximum number of `Byte`s which may be uploaded.
+     */
+    init {
+        Validate.notNull(mongoClient)
+        Validate.isTrue(payloadLimit > 0)
+        this.mongoClient = mongoClient
+        this.payloadLimit = payloadLimit
+    }
+
+    override fun handleAuthorizedRequest(
+        ctx: RoutingContext, users: Set<User>,
+        header: MultiMap
+    ) {
+        LOGGER.info("Received new measurement request.")
+        val request = ctx.request()
+        val session = ctx.session()
+        try {
+            // Load authenticated user
+            val username = ctx.user().principal().getString("username")
+            val matched = users.stream().filter { u: User -> u.name == username }.collect(Collectors.toList())
+            Validate.isTrue(matched.size == 1)
+            val loggedInUser = matched.stream().findFirst().get() // Make sure it's the matched user
+            val bodySize = PreRequestHandler.bodySize(request.headers(), payloadLimit, "content-length")
+            val metaData = metaData(request)
+            checkSessionValidity(session, metaData)
+
+            // Handle upload status request
+            if (bodySize == 0L) {
+                StatusHandler(mongoClient).handle(ctx)
+                return
+            }
+
+            // Handle first chunk
+            val contentRange = contentRange(request, bodySize)
+            if (session.get<Any?>(UPLOAD_PATH_FIELD) == null) {
+                handleFirstChunkUpload(ctx, request, session, loggedInUser, contentRange, metaData, null)
+                return
+            }
+
+            // Search for previous upload chunk
+            val fs = ctx.vertx().fileSystem()
+            val path = session.get<Any>(UPLOAD_PATH_FIELD) as String
+            request.pause()
+            fs.exists(path).onSuccess { fileExists: Boolean? ->
+                try {
+                    request.resume()
+                    if (!fileExists!!) {
+                        session.remove<Any>(UPLOAD_PATH_FIELD) // was linked to non-existing file
+                        handleFirstChunkUpload(ctx, request, session, loggedInUser, contentRange, metaData, path)
+                        return@onSuccess
```

Comment:
Just out of interest, was means this `@onSuccess` ? Just documentation?

Reply:
I have no idea. This code was just converted from Java automatically by IntelliJ. I only checked that it compiles and builds. I'll have to rip out almost everything from here anyways for the new Implementation with Picture support, so I did not bother to fiddle around with this stuff too much.